vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/json/LibjnDescendantArraysScalarEvaluatorFactory.java

```diff
@@ -109,23 +119,23 @@ public void appendSequence(TaggedValuePointable tvp, ArrayPointable ap) throws S
             public void insideObject(TaggedValuePointable tvp) throws SystemException {
                 boolean inObject = false;
                 tvp.getValue(op);
-                TaggedValuePointable tempTvp1 = ppool.takeOne(TaggedValuePointable.class);
+                TaggedValuePointable tempTvp = ppool.takeOne(TaggedValuePointable.class);
                 try {
                     op.getKeys(tvp);
                     tvp.getValue(stringp);
-                    op.getValue(stringp, tempTvp1);
+                    op.getValue(stringp, tempTvp);
                 } catch (IOException e1) {
                     throw new SystemException(ErrorCode.SYSE0001, e1);
                 }
-                if (tempTvp1.getTag() == ValueTag.OBJECT_TAG) {
+                if (tempTvp.getTag() == ValueTag.OBJECT_TAG) {
                     inObject = true;
                 }
-                ppool.giveBack(tempTvp1);
                 if (inObject) {
```

Comment:
It seems like you don't need the variable inObject. Instead just use the if statement above (tempTvp.getTag() == ValueTag.OBJECT_TAG)